File: api/DTOs/HR/CandidateAssessmentDto.cs
namespace api.DTOs.HR
{
    public class 
    CandidateAssessmentDto
    {
        public int Id { get; set; }
        public int CandidateId { get; set; }
        public int ApplicationNo { get; set; }
        public string CandidateName { get; set; }
        public string CustomerName { get; set; }    
        public int OrderItemId { get; set; }
        public string CategoryRef { get; set; }
        public int OrderId { get; set; }
        public string ProfessionName { get; set; }
        public string  RequireInternalReview { get; set; }
        public string AssessedByUsername { get; set; }
        public string AssessResult { get; set; }
        public DateTime AssessedOn { get; set; }
        public int ChecklistHRId { get; set; }  
        public string ChecklistedByName { get; set; }   
        public DateTime ChecklistedOn { get; set; }
        public ICollection<AssessmentItemDto> AssessmentItemsDto {get; set;}
    }

    public class AssessmentItemDto
    {
        public int Id { get; set; }
        public int CandidateAssessmentId { get; set; }
        public int QuestionNo { get; set; }
        public string Question { get; set; }
        public bool IsMandatory { get; set; }   
        public int MaxPoints { get; set; }
        public int Points { get; set; }
        public string Remarks { get; set; }
    }
}

File: client/src/app/_dtos/hr/candidateAssessmentDto.ts
export interface ICandidateAssessmentDto
{
     id: number;
     candidateId: number;
     candidateName: string;
     applicationNo: number;

     customerName: string;
     professionName: string;
     categoryRefAndName: string;
     orderDate: Date;

     orderItemId: number;
     categoryRef: string;
     orderId: number;
     //professionName: string;
     requireInternalReview: boolean;
     assessedByUsername: string;
     assessResult: string;
     assessedOn: Date;
     checklistHRId: number;
     checklistedByName: string;
     checklistedOn: Date;
    
     assessmentItemsDto: IAssessmentItemDto[]
}

export class CandidateAssessmentDto implements ICandidateAssessmentDto
{
     id= 0;
     candidateId= 0;
     candidateName= '';
     applicationNo= 0;

     customerName= '';
     professionName= '';
     categoryRefAndName= '';
     orderDate = new Date;

     orderItemId= 0;
     categoryRef= '';
     orderId= 0;
     //professionName= '';
     requireInternalReview = false;
     assessedByUsername= '';
     assessResult= '';
     assessedOn = new Date;
     checklistHRId= 0;
     checklistedByName= '';
     checklistedOn = new Date;
    
     assessmentItemsDto: IAssessmentItemDto[] = [];
}


export interface IAssessmentItemDto
{
     id: number;
     candidateAssessmentId: number;
     questionNo: number;
     isMandatory: boolean;
     assessed: boolean;
     question: string;
     maxPoints: number;
     points: number;
     remarks: string;
}

export class AssessmentItemDto implements IAssessmentItemDto
{
     id = 0;
     candidateAssessmentId = 0;
     questionNo=0;
     isMandatory = false;
     assessed = false;
     question = '';
     maxPoints = 0;
     points = 0;
     remarks = '';
}